### PR TITLE
Eager 760

### PR DESCRIPTION
**Pull request by zumic96**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes [#760](https://drivevariant.atlassian.net/browse/EAGER-760)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

```text
Ran multiple tests to verify each feature works as expected.
All tests were run locally on laptop on Variant-Playground account.
```

- [ ] Sanity Testing

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added documentation to test

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -37,54 +62,153 @@ resource "aws_s3_bucket" "bucket" {
     }
   }
 
+  dynamic "replication_configuration" {
```
Comment (vibindaniel) [inline main.tf:65]: remove this pls

File: main.tf [hunk 2]
```diff
@@ -37,54 +62,153 @@ resource "aws_s3_bucket" "bucket" {
...
 
   versioning {
-    enabled = true
+    enabled = lookup(var.versioning, "enabled", true)
+    mfa_delete = lookup(var.versioning, "mfa_delete", false)
   }
 
 }
 
 resource "aws_s3_bucket_public_access_block" "bucket" {
```
Comment (vibindaniel) [inline main.tf:101]: revert this pls

File: main.tf [hunk 3]
```diff
@@ -37,54 +62,153 @@ resource "aws_s3_bucket" "bucket" {
...
           }
         }
       }
-    ]
+    ],
+    local.policy)
   })
 }
+
+resource "aws_s3_bucket_object" "object_staging" {
```
Comment (vibindaniel) [inline main.tf:140]: remove everything from here to end of the file

**Review by vibindaniel — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (zumic96) [reply main.tf]: removed

File: main.tf [hunk 4]
```diff
@@ -1,17 +1,24 @@
...
 resource "aws_s3_bucket" "bucket" {
   #ts:skip=AWS.S3Bucket.LM.MEDIUM.0078 need to skip this rule
 
   bucket_prefix = var.bucket_prefix
-  acl           = "private"
+  acl           = var.acl
 
   force_destroy = var.force_destroy
 
-  lifecycle {
```
[the remark this answers is not in the document]
Comment (zumic96) [reply main.tf:9]: reverted

File: main.tf [hunk 5]
```diff
@@ -1,17 +1,24 @@
...
+  #   ]
+}
+data "aws_caller_identity" "current" {}
+
 resource "aws_s3_bucket" "bucket" {
   #ts:skip=AWS.S3Bucket.LM.MEDIUM.0078 need to skip this rule
 
   bucket_prefix = var.bucket_prefix
-  acl           = "private"
+  acl           = var.acl
```
[the remark this answers is not in the document]
Comment (zumic96) [reply main.tf]: reverted

File: main.tf [hunk 6]
```diff
@@ -37,54 +62,153 @@ resource "aws_s3_bucket" "bucket" {
...
       apply_server_side_encryption_by_default {
-        sse_algorithm = "AES256"
+        sse_algorithm = lookup(var.server_side_encryption_configuration["rule"]["apply_server_side_encryption_by_default"], "sse_algorythm", "AES256")
+        kms_master_key_id = lookup(var.server_side_encryption_configuration["rule"]["apply_server_side_encryption_by_default"], "sse_algorythm", "AES256") != "AES256" ? lookup(var.server_side_encryption_configuration["rule"]["apply_server_side_encryption_by_default"], "kms_master_key_id", "aws/s3") : null
       }
+      bucket_key_enabled = lookup(var.server_side_encryption_configuration["rule"], "bucket_key_enabled", false)
     }
   }
 
   versioning {
```
[the remark this answers is not in the document]
Comment (zumic96) [reply main.tf]: reverted

File: main.tf [hunk 7]
```diff
@@ -37,54 +62,153 @@ resource "aws_s3_bucket" "bucket" {
...
+          prefix = lookup(lookup(lookup(replication_configuration.value, "rules", {}), "filter", {}), "prefix", null)
+        }
+        status = lookup(lookup(replication_configuration.value, "rules", {}), "status", null)
+      }
+    }
+  }
+
   server_side_encryption_configuration {
     rule {
       apply_server_side_encryption_by_default {
```
[the remark this answers is not in the document]
Comment (zumic96) [reply main.tf]: reverted

File: main.tf (hunk 2 above)
Comment (zumic96) [reply main.tf]: reverted

File: main.tf (hunk 3 above)
Comment (zumic96) [reply main.tf]: removed

**Review by vibindaniel — DISMISSED** [review]

**Comment by zumic96** [comment]
will make new PR with signed commits, sry for spam

**Pull request closed without merging**